src/cmd-koji-upload

```diff
@@ -618,56 +728,95 @@ Environment variables are supported:
-    parser.add_argument("--build", default="latest",
+    parent_parser.add_argument("--build", default="latest",
                         help="Override build id, defaults to latest")
-    parser.add_argument("--buildroot", default="builds",
+    parent_parser.add_argument("--buildroot", default="builds",
                         help="Build diretory")
-    parser.add_argument("--dump", default=False, action='store_true',
+    parent_parser.add_argument("--dump", default=False, action='store_true',
                         help="Dump the manfiest and exit")
-    parser.add_argument("--no-upload", default=False, action='store_true',
+    parent_parser.add_argument("--no-upload", default=False, action='store_true',
                         help="Do not upload, just parse the build")
-    parser.add_argument("--arch", default=get_basearch(),
+    parent_parser.add_argument("--arch", default=get_basearch(),
                         help="Set the build architecture")
 
     # Koji specific options
-    parser.add_argument("--no-auth", action='store_false', dest="auth",
+    parent_parser.add_argument("--no-auth", action='store_false', dest="auth",
                         help="Skip Kerberos auth, use if already auth'd")
-    parser.add_argument("--keytab",
+    parent_parser.add_argument("--keytab",
                         default=os.environ.get("KOJI_KEYTAB", None),
                         help="location of the keytab file to use for auth")
-    parser.add_argument('--owner', required=True,
+    parent_parser.add_argument('--owner', required=False,
                         default=os.environ.get("KOJI_USERNAME", None),
                         help='koji user name that owns this build')
-    parser.add_argument('--tag', required=True,
-                        default=os.environ.get("KOJI_TAG", None),
-                        help='tag this build, eg. awesome-candidate')
-    parser.add_argument('--profile', required=True,
+    # REQUIRED for session
+    parent_parser.add_argument('--profile', required=False,
                         default=os.environ.get("KOJI_PROFILE", None),
                         help='profile to use, e.g. prod, stage, test')
-    args = parser.parse_args()
 
+    sub_commands = parser.add_subparsers(
+        dest="_command",
+        help="Other commands")
+
+    upload_cmd = sub_commands.add_parser(
+        "upload", help="Uploads to koji", parents=[parent_parser], add_help=False)
+
+    sub_commands.add_parser(
+        "reserve-id", help="Reserves a koji id", parents=[parent_parser], add_help=False)
+
+    upload_cmd.add_argument(
+        '--tag', required=True,
+        default=os.environ.get("KOJI_TAG", None),
+        help='tag this build, eg. awesome-candidate')
+
+    upload_cmd.add_argument(
+        '--reserve-id', required=False, action='store_true',
+        help='Creates a reservation ID in the upload process')
+
+    upload_cmd.add_argument(
+        '--load-id', required=False,
```

Comment:
Perhaps this option should be renamed?  It strikes me as a boolean or maybe an int that is expected.

Could you just call it `reserve-id-file` to make it consistent with the rest of the cod?

Reply:
Or maybe `reserve-id-state-file` if you want to be really precise